hoomd/hpmc/PairPotential.h

```diff
@@ -0,0 +1,148 @@
+        {
+        return m_r_cut_squared_total[m_type_param_index(type_i, type_j)];
+        }
+
+    /*** Evaluate the energy of the pair interaction
+
+        energy is given a pre-computed r_squared as many potentials use this parameter and the
+        caller has already computed it.
+
+        @param r_squared Pre-computed dot(r_ij, r_ij).
+        @param r_ij Vector pointing from particle i to j.
+        @param type_i Integer type index of particle i.
+        @param charge_i Charge of particle i.
+        @param q_i Orientation quaternion of particle i.
+        @param type_j Integer type index of particle j.
+        @param q_j Orientation quaternion of particle j.
+        @param charge_j Charge of particle j.
+        @returns Energy of the pair interaction.
+    */
+    virtual LongReal energy(const LongReal r_squared,
+                            const vec3<LongReal>& r_ij,
+                            const unsigned int type_i,
+                            const quat<LongReal>& q_i,
+                            const LongReal charge_i,
+                            const unsigned int type_j,
+                            const quat<LongReal>& q_j,
+                            const LongReal charge_j) const
+        {
+        return 0;
+        }
+
+    protected:
+    /// The system definition.
+    std::shared_ptr<SystemDefinition> m_sysdef;
+
+    /// Indexer to access arrays by pairs of type parameters
+    Index2D m_type_param_index;
+
+    /// Set the non-additive cutoff radius for both i,j and j,i pairs.
+    void setRCutNonAdditive(unsigned int type_i, unsigned int type_j, LongReal r)
+        {
+        m_r_cut_non_additive[m_type_param_index(type_i, type_j)] = r;
+        m_r_cut_non_additive[m_type_param_index(type_j, type_i)] = r;
+
+        updateRCutTotal(type_i, type_j);
+        }
+
+    /// Returns the additive part of the cutoff distance for a given type.
+    void setRCutAdditive(unsigned int type, LongReal r)
+        {
+        m_r_cut_additive[type] = r;
+
+        for (unsigned int type_j = 0; type_j < m_sysdef->getParticleData()->getNTypes(); type_j++)
+            {
+            updateRCutTotal(type, type_j);
+            }
+        }
+
+    private:
+    /// The non additive r_cut matrix (indexed by m_type_param_index).
+    std::vector<LongReal> m_r_cut_non_additive;
+
+    /// The additive r_cut values (indexed by type).
+    std::vector<LongReal> m_r_cut_additive;
+
+    /// Pre computed total r_cut squared (indexed by m_type_param_index).
+    std::vector<LongReal> m_r_cut_squared_total;
+
+    /// Update the pre-computed r_cut_total value.
+    void updateRCutTotal(unsigned int type_i, unsigned int type_j)
+        {
+        unsigned int type_pair = m_type_param_index(type_i, type_j);
+        LongReal r_cut = m_r_cut_non_additive[type_pair] + LongReal(0.5) * m_r_cut_additive[type_i]
+                         + m_r_cut_additive[type_j];
```

Comment:
Is this meant to be?
```suggestion
        LongReal r_cut = m_r_cut_non_additive[type_pair] + LongReal(0.5) * (m_r_cut_additive[type_i]
                         + m_r_cut_additive[type_j]);
```

Reply:
Yes, good catch!